posthog/temporal/batch_exports/transformer.py

```diff
@@ -167,6 +195,54 @@ def _write(self, content: bytes | str):
         compressed_content = self.compress(content)
         yield compressed_content
 
+    async def iter_transformed_record_batches(
+        self, record_batches: collections.abc.AsyncIterator[pa.RecordBatch], max_file_size_bytes: int = 0
+    ) -> collections.abc.AsyncIterator[bytes]:
+        """Distribute transformation of record batches into multiple processes.
+
+        This is only supported for non-brotli compressed batch exports, so we
+        default to the parent implementation when using brotli compression.
+        """
+        if self.compression == "brotli":
+            async for chunk in super().iter_transformed_record_batches(record_batches, max_file_size_bytes):
+                yield chunk
+            return
+
+        with concurrent.futures.ProcessPoolExecutor() as executor:
+            loop = asyncio.get_running_loop()
+            tasks = []
+
+            async for record_batch in record_batches:
+                sink = pa.MockOutputStream()
+                with pa.ipc.new_stream(sink, record_batch.schema) as writer:
+                    writer.write_batch(record_batch)
+                size = sink.size()
+
+                shm = sm.SharedMemory(create=True, size=size)
+
+                try:
+                    stream = pa.FixedSizeBufferWriter(pa.py_buffer(shm.buf))
+                    with pa.RecordBatchStreamWriter(stream, record_batch.schema) as writer:
+                        writer.write_batch(record_batch)
+
+                    tasks.append(
+                        loop.run_in_executor(executor, transform_record_batch_from_shared_memory, shm.name, self)
+                    )
+
+                    del stream
+                    del writer
+                    del record_batch
+                    del sink
+
+                finally:
+                    shm.close()
```

Comment:
would this not close the shared memory before `transform_record_batch_from_shared_memory` has finished running?